### PR TITLE
added project site navigation for ROs

### PR DESCRIPTION
**Pull request by ankitecd**
Added support for ROs (TPOs) to navigate between their projects and sites on the map

### CODE REVIEW

File: app/components/MainScreen/ProjectAndSiteSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  const setSelectedProjectStates = (sites: any, project: any) => {
+    setProjectSites(sites);
+    if (sites && sites.length > 0) {
+      setSelectedProjectSiteId(sites[0].value);
+      setSiteCenterCoordinate([]);
+
+      setSiteBounds(bbox(sites[0].geometry));
+    } else {
+      setSelectedProjectSiteId(null);
+      setSiteBounds([]);
```
Comment (norbertschuler) [inline app/components/MainScreen/ProjectAndSiteSelector.tsx:84]: Got a crash here:
Simulator Screen Shot - iPhone 13 - 2021-12-01 at 12 54 17

File: app/components/MainScreen/ProjectAndSiteSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      setSiteCenterCoordinate(turfCenter(JSON.parse(project.geometry)).geometry.coordinates);
```
Comment (norbertschuler) [inline app/components/MainScreen/ProjectAndSiteSelector.tsx:85]: The code also crash here with a JSON.parse error as project.geometry was undefined.

**Review by norbertschuler — CHANGES_REQUESTED** [review]
Starting the app with existing data before, produces the above errors. If you start with a clean app, the native part of MapBox code crashes with:

```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'A multipoint must have at least one vertex.'
abort() called
```

**Comment by ankitecd** [comment]
@norbertschuler I tested it on iOS and Android, it's working for me. Can you check again and let me know if it's working for you?

**Comment by norbertschuler** [comment]
- [ ] The box showing "pending uploads" is not rendered very nice any more:

Simulator Screen Shot - iPhone 13 - 2021-12-17 at 16 47 31

**Review by norbertschuler — APPROVED** [review]
PR works for me. Now if you have chosen a project on the map, you just wonder why the chosen project is not automatically used for new registrations. That would be a future enhancement maybe? But as the chosen project also is not made persistent anyway (you always get set back to your first project after killing the app), it does not have priority for me.

**Pull request merged**